src/main/java/ch/njol/skript/effects/EffWakeupSleep.java

```diff
@@ -0,0 +1,130 @@
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.config.Node;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.SyntaxStringBuilder;
+import ch.njol.skript.util.Direction;
+import ch.njol.util.Kleenean;
+import org.bukkit.Location;
+import org.bukkit.entity.*;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.log.runtime.SyntaxRuntimeErrorProducer;
+
+@Name("Wake And Sleep")
+@Description({
+	"Make bats and foxes sleep or wake up.",
+	"Make villagers sleep by providing a location of a bed.",
+	"Make players sleep by providing a location of a bed and 'with force' to bypass nearby monsters.",
+	"Using 'without spawn location update' will make players wake up without setting their spawn location to the bed."
+})
+@Examples({
+	"make {_fox} go to sleep",
+	"make {_bat} stop sleeping",
+	"make {_villager} start sleeping at location(0, 0, 0)",
+	"make player go to sleep at location(0, 0, 0) with force",
+	"make player wake up without spawn location update"
+})
+@Since("INSERT VERSION")
+public class EffWakeupSleep extends Effect implements SyntaxRuntimeErrorProducer {
+
+	static {
+		Skript.registerEffect(EffWakeupSleep.class,
+			"make %livingentities% (start sleeping|[go to] sleep) [%-direction% %-location%]",
+			"force %livingentities% to (start sleeping|[go to] sleep) [%-direction% %-location%]",
+			"make %players% (start sleeping|[go to] sleep) %direction% %location% [force:with force]",
+			"force %players% to (start sleeping|[go to] sleep) %direction% %location% [force:with force]",
```

Comment:
These patterns conflict, the latter will never be used unless `with force` is added.